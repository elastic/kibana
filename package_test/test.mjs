/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import packages from './search.json' assert { type: 'json' };
import semver from 'semver';
import axios from 'axios';
import fs from 'fs';
import path from 'path';

const headers = {
  'Content-Type': 'application/json',
  Accept: 'application/json',
  'kbn-xsrf': 'f62a882f-7df0-4cf2-bd46-13566b795301',
  Authorization: 'Basic ZWxhc3RpYzpjaGFuZ2VtZQ==',
};

const installPackage = async (version) => {
  let data = JSON.stringify({
    force: true,
  });

  return axios.request({
    method: 'post',
    maxBodyLength: Infinity,
    url: `http://localhost:5601/kbn/api/fleet/epm/packages/security_detection_engine/${version}`,
    headers: {
      ...headers,
      'elastic-api-version': '2023-10-31',
    },
    data,
  });
};

const installAllRules = async () => {
  let data = JSON.stringify({
    mode: 'ALL_RULES',
  });

  let config = {
    method: 'post',
    maxBodyLength: Infinity,
    url: 'http://localhost:5601/kbn/internal/detection_engine/prebuilt_rules/installation/_perform',
    headers: {
      ...headers,
      'elastic-api-version': '1',
    },
    data,
  };

  return axios.request(config);
};

const reviewUpgrade = async () => {
  let config = {
    method: 'post',
    maxBodyLength: Infinity,
    url: 'http://localhost:5601/kbn/internal/detection_engine/prebuilt_rules/upgrade/_review',
    headers: {
      ...headers,
      'elastic-api-version': '1',
    },
  };

  return axios.request(config);
};

const performUpgrade = async () => {
  let data = JSON.stringify({
    mode: 'ALL_RULES',
  });

  let config = {
    method: 'post',
    maxBodyLength: Infinity,
    url: 'http://localhost:5601/kbn/internal/detection_engine/prebuilt_rules/upgrade/_perform',
    headers: {
      ...headers,
      'elastic-api-version': '1',
    },
    data: data,
  };

  return axios.request(config);
};

const deleteAllRules = async () => {
  let data = JSON.stringify({
    action: 'delete',
    query: '',
  });

  let config = {
    method: 'post',
    maxBodyLength: Infinity,
    url: 'http://localhost:5601/kbn/api/detection_engine/rules/_bulk_action?dry_run=false',
    headers: {
      ...headers,
      'elastic-api-version': '2023-10-31',
    },
    data: data,
  };

  return axios.request(config);
};

const filtered = packages
  .filter((pkg) => pkg.conditions.kibana.version.startsWith('^8'))
  .sort((a, b) => semver.compare(a.version, b.version));

const resultsDir = path.resolve('./results');
if (!fs.existsSync(resultsDir)) {
  fs.mkdirSync(resultsDir);
}

for (const pkg of filtered) {
  const { version } = pkg;
  const performFilePath = path.join(resultsDir, `${version}_perform.json`);

  if (fs.existsSync(performFilePath)) {
    console.log(`Skipping version ${version} as it has already been processed.`);
    continue;
  }

  try {
    console.log('Deleting all rules');
    await deleteAllRules();

    console.log(`Installing package ${pkg.name} version ${version}`);
    await installPackage(version);

    console.log(`Installing all rules`);
    const installAllRulesResponse = await installAllRules();
    const installAllRulesData = JSON.stringify(installAllRulesResponse.data, null, 2);
    fs.writeFileSync(path.join(resultsDir, `${version}_install.json`), installAllRulesData);

    console.log(`Installing the latest package`);
    await installPackage('8.17.1');

    console.log('Reviewing upgrade');
    const reviewResponse = await reviewUpgrade();
    const reviewData = JSON.stringify(reviewResponse.data, null, 2);
    fs.writeFileSync(path.join(resultsDir, `${version}_review.json`), reviewData);

    console.log('Performing upgrade');
    const performResponse = await performUpgrade();
    const performData = JSON.stringify(performResponse.data, null, 2);
    fs.writeFileSync(performFilePath, performData);
  } catch (error) {
    console.error(`Error processing version ${version}:`, error);
  }
}
