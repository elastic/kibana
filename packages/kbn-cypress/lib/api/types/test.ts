export enum TestState {
  Failed = "failed",
  Passed = "passed",
  Pending = "pending",
  Skipped = "skipped",
}

export interface TestAttempt {
  state: TestState;
  error: CypressCommandLine.TestError | null;
  wallClockStartedAt: string | null;
  wallClockDuration: number | null;
  videoTimestamp: number | null;
}

interface TestConfig {
  retries:
    | {
        openMode: number;
        runMode: number;
      }
    | number;
}

export interface TestHook {
  clientId: string;
  type: "before each";
  title: string[];
  body: string;
}

export interface Test {
  state: TestState;
  testId: string;
  displayError: string | null;
  title: string[];
  config?: null | TestConfig;
  hookIds: string[];
  body: string;
  attempts: TestAttempt[];
  hooks: TestHook[] | null;
}

export type SetTestsPayload = Pick<
  Test,
  "body" | "title" | "config" | "hookIds"
> & { clientId: string };
