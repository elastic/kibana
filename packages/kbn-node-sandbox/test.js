'use strict';
const assert = require('assert');
const sandbox = require("./index.js");
const child_process = require("child_process");

function tryExec() {
  // child_process.spawn() throws an exception immediately if it fails.
  // so there's no need to bind to the 'error' event.
  child_process.exec("true");
}

assert.doesNotThrow(tryExec);

let s = sandbox.activate()
assert.ok(s.success, "Sandbox failed to activate: " + s.message);

// EACCES on Linux, UNKNOWN on Windows.
assert.throws(tryExec, /Error: spawn (EACCES|UNKNOWN)/);
