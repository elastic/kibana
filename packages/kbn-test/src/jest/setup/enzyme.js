/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
