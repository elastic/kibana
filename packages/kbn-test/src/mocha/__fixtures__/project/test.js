/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

describe('SUITE', () => {
  it('works', () => {});
  it('fails', () => {
    throw new Error('FORCE_TEST_FAIL');
  });

  describe('SUB_SUITE', () => {
    beforeEach('success hook', () => {});
    beforeEach('fail hook', () => {
      throw new Error('FORCE_HOOK_FAIL');
    });

    it('never runs', () => {});
  });
});
