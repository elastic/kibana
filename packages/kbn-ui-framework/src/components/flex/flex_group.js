/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const gutterSizeToClassNameMap = {
  none: '',
  small: 'kuiFlexGroup--gutterSmall',
  medium: 'kuiFlexGroup--gutterMedium',
  large: 'kuiFlexGroup--gutterLarge',
  extraLarge: 'kuiFlexGroup--gutterExtraLarge',
};

export const GUTTER_SIZES = Object.keys(gutterSizeToClassNameMap);

const alignItemsToClassNameMap = {
  stretch: '',
  flexStart: 'kuiFlexGroup--alignItemsStart',
  flexEnd: 'kuiFlexGroup--alignItemsEnd',
  center: 'kuiFlexGroup--alignItemsCenter',
};

export const ALIGN_ITEMS = Object.keys(alignItemsToClassNameMap);

const justifyContentToClassNameMap = {
  flexStart: '',
  flexEnd: 'kuiFlexGroup--justifyContentFlexEnd',
  center: 'kuiFlexGroup--justifyContentCenter',
  spaceBetween: 'kuiFlexGroup--justifyContentSpaceBetween',
  spaceAround: 'kuiFlexGroup--justifyContentSpaceAround',
  spaceEvenly: 'kuiFlexGroup--justifyContentSpaceEvenly',
};

export const JUSTIFY_CONTENTS = Object.keys(justifyContentToClassNameMap);

export const KuiFlexGroup = ({
  children,
  className,
  gutterSize,
  alignItems,
  justifyContent,
  wrap,
  ...rest
}) => {
  const classes = classNames(
    'kuiFlexGroup',
    gutterSizeToClassNameMap[gutterSize],
    alignItemsToClassNameMap[alignItems],
    justifyContentToClassNameMap[justifyContent],
    className,
    {
      'kuiFlexGroup--wrap': wrap,
    },
  );

  return (
    <div
      className={classes}
      {...rest}
    >
      {children}
    </div>
  );
};

KuiFlexGroup.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  gutterSize: PropTypes.oneOf(GUTTER_SIZES),
  alignItems: PropTypes.oneOf(ALIGN_ITEMS),
  justifyContent: PropTypes.oneOf(JUSTIFY_CONTENTS),
  wrap: PropTypes.bool,
};

KuiFlexGroup.defaultProps = {
  gutterSize: 'large',
  alignItems: 'stretch',
  justifyContent: 'flexStart',
  wrap: false,
};
