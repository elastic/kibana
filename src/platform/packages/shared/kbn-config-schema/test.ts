import { TypeOf, schema } from '@kbn/config-schema';
import _ from 'lodash';

const testSchemaObject = schema.object({
    a: schema.string(),
    b: schema.number({ defaultValue: 5 }),
    c: schema.maybe(schema.boolean()),
});

const testSchemaAllOf = schema.allOf([
    testSchemaObject,
    schema.object({
        d: schema.string(),
        e: schema.number({ defaultValue: 3 }),
        f: schema.maybe(schema.boolean()) 
    }),
])

const testSchemaObjectInput = testSchemaObject.getInputSchema();
const testSchemaAllOfInput = testSchemaAllOf.getInputSchema();

type TestSchemaObjectInputType = typeof testSchemaObjectInput.type;
type TestSchemaAllOfInputType = typeof testSchemaAllOfInput.type;

const testObject: TestSchemaObjectInputType = {
    a: 'hello',
    // b is optional because it has a default value
    // c is optional because it's maybe
};

// complains that b is missing
