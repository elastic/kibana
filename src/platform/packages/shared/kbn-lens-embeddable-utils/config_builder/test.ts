import type { MetricState } from "./schema";

const test: MetricState = {
  type: 'metric',
  dataset: {
    type: 'index',
    index: 'test',
    time_field: 'test'
  },
  metric: {
    operation: 'count',

  },
  secondary_metric: {
    operation: 'min',
    field: 'test',
  },
  breakdown_by: {
    operation: 'terms',
    fields: ['test'],
    size: 5,
    increase_accuracy: true,
    collapse_by: 'avg',
  },
  ignore_global_filters: true,
  samplings: 1,
}