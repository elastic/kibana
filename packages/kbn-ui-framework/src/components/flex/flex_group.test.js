import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { KuiFlexGroup } from './flex_group';

describe('KuiFlexGroup', () => {
  test('is rendered', () => {
    const component = render(
      <KuiFlexGroup {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
