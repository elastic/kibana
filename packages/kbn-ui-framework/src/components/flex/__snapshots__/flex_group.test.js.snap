// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KuiFlexGroup is rendered 1`] = `
<div
  aria-label="aria-label"
  class="kuiFlexGroup kuiFlexGroup--gutterLarge testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;
